src/plugins/kibana/index.js

```diff
@@ -22,7 +22,8 @@ module.exports = function (kibana) {
           'spyModes',
           'fieldFormats',
           'navbarExtensions',
-          'settingsSections'
+          'settingsSections',
+          'docTableDetailViews'
```

Comment:
I like this as an extension point for plugins, but I think the existing doc viewers should use it (rather than being [included directly](https://github.com/trevan/kibana/blob/doc-detail-views-registry/src/ui/public/doc_viewer/doc_viewer.js#L5-L6)). Perhaps they should get their own plugin, similar to the `kbn_vislib_vis_types` plugin.

It may make sense to go generic with this as well, and just call them `docViews`, then the `docViewer` (or any other component that wants to display docs) can utilize them?